python/cuda_parallel/cuda/parallel/experimental/algorithms/reduce.py

```diff
@@ -64,17 +65,18 @@ def __init__(
         cc_major, cc_minor = cuda.get_current_device().compute_capability
         cub_path, thrust_path, libcudacxx_path, cuda_include_path = get_paths()
         bindings = get_bindings()
-        self.op_wrapper = _Op(h_init.dtype, op)
+        self.op_wrapper = _Op(h_init, op)
         d_out_cccl = cccl.to_cccl_iter(d_out)
         self.build_result = cccl.DeviceReduceBuildResult()
 
-        # TODO Figure out caching
         error = bindings.cccl_device_reduce_build(
             ctypes.byref(self.build_result),
             d_in_cccl,
             d_out_cccl,
             self.op_wrapper.handle(),
-            cccl.host_array_to_value(h_init),
+            cccl.to_cccl_value(
+                h_init if isinstance(h_init, np.ndarray) else h_init._data
+            ),
```

Comment:
Minor point, but I feel like it would be more readable to extract this expression to a variable and pass that in.

Reply:
Given @rwgk's concerns here as well (https://github.com/NVIDIA/cccl/pull/3218#discussion_r1913504055), I've made it so that `to_cccl_value()` accepts a `GpuStruct` in addition to `ndarray`:

https://github.com/NVIDIA/cccl/pull/3218/commits/e75bc7c0c1e2cf36356459593268edbb1c8eeb78